### ISSUE TITLE
Pag 29 - Error during Building a toochain for QEMU

### ISSUE THREAD

**Issue by fernandoBranqs** [PacktPublishing/Mastering-Embedded-Linux-Programming-Third-Edition#76]
Pag 29 - Error during Building a toochain for QEMU after the command:

bin/ct-ng build

[INFO ]  Installing MPC for host: done in 8.79s (at 04:47)
[INFO ]  =================================================================
[INFO ]  Installing binutils for host
[EXTRA]    Configuring binutils
[EXTRA]    Building binutils
[ERROR]    make[2]: *** [Makefile:7650: configure-ld] Error 1
[ERROR]    make[2]: *** Waiting for unfinished jobs....
[ERROR]    make[1]: *** [Makefile:1034: all] Error 2
[ERROR]   
[ERROR]  >>
[ERROR]  >>  Build failed in step 'Installing binutils for host'
[ERROR]  >>        called in step '(top-level)'
[ERROR]  >>
[ERROR]  >>  Error happened in: CT_DoExecLog[scripts/functions@377]
[ERROR]  >>        called from: do_binutils_backend[scripts/build/binutils/binutils.sh@214]
[ERROR]  >>        called from: do_binutils_for_host[scripts/build/binutils/binutils.sh@70]
[ERROR]  >>        called from: main[scripts/crosstool-NG.sh@708]
[ERROR]  >>
[ERROR]  >>  For more info on this error, look at the file: 'build.log'
[ERROR]  >>  There is a list of known issues, some with workarounds, in:
[ERROR]  >>      https://crosstool-ng.github.io/docs/known-issues/
[ERROR]  >>
[ERROR]  >>  If you feel this is a bug in crosstool-NG, report it at:
[ERROR]  >>      https://github.com/crosstool-ng/crosstool-ng/issues/
[ERROR]  >>
[ERROR]  >>  Make sure your report includes all the information pertinent to this issue.
[ERROR]  >>  Read the bug reporting guidelines here:
[ERROR]  >>      http://crosstool-ng.github.io/support/
[ERROR]   
[ERROR]  (elapsed: 5:08.38)
[05:08] / make: *** [bin/ct-ng:261: build] Error 2

I'm using a real machine with 20.04 Desktop Ubuntu. I have tried on VMs also, and i had the same error.

Could anyone help me ?

**Comment by fvasquez** [comment]
Looks like permissions. Check your `build.log` file and see the following issue.

https://github.com/crosstool-ng/crosstool-ng/issues/1607

**Comment by fernandoBranqs** [comment]
Thank you very much for your help. 

But something a little strange happened. Before I saw your answer, I made an attempt to create the toolchain for BeagleBone. Everything worked. 

So after seeing your message, I tried again to create the toolchain for QEMU to be able to analyze the build.log file. Unexpectedly, it now worked.

 It's as if creating the toolchain for BeagleBone solved the problem that was occurring with QEMU. Very strange, but now it's working. 
 
 Once again, thank you very much for your help.
